### PR TITLE
chore(schema): support spice type data set

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 1]
```diff
@@ -387,7 +388,7 @@ export const createDataSet = async (quickSight: QuickSight, awsAccountId: string
       AwsAccountId: awsAccountId,
       DataSetId: datasetId,
       Name: datasetId,
-      ImportMode: props.importMode,
+      ImportMode: props.useSpice === 'yes' ? DataSetImportMode.SPICE : DataSetImportMode.DIRECT_QUERY,
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:391]: defining `useSpice` as boolean is simpler

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 2]
```diff
@@ -102,6 +103,9 @@ export const handler = async (event: ResourceEvent, _context: Context): Promise<
   }
   const timezoneDict = timezoneJsonArrayToDict(JSON.parse(timezone));
 
+  logger.info('useSpice:', props.useSpice);
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:106]: you could use powertools's logger feature, 

```ts
logger.info('dataset info': { useSpice: props.useSpice,
  dataSets: props.dashboardDefProps.dataSets });
```

File: src/reporting/private/template-def-spice.json
Comment (zxkane) [inline src/reporting/private/template-def-spice.json:1]: If there are just a few differences between the spice and non-spice templates, could we use mustache to render the different parts?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:391]: Its value comes from a CFN parameter, defining it as a boolean isn't very convenient.

File: src/reporting/private/template-def-spice.json
Comment (llmin) [reply src/reporting/private/template-def-spice.json:1]: many difference including positions of visuals.

**Review by zxkane — APPROVED** [review]

File: src/reporting/quicksight-custom-resource.ts [hunk 3]
```diff
@@ -104,62 +104,817 @@ export function createQuicksightCustomResource(
     templateId: props.templateId,
     dataSourceArn: props.dataSourceArn,
     databaseName: databaseName,
-    dataSets: [
-      //Base View
+    dataSets: _getDataSetDefs('no', eventViewColumns, eventViewProjectedColumns, tenYearsAgo, futureDate),
+    dataSetsSpice: _getDataSetDefs('yes', eventViewColumns, eventViewProjectedColumns, tenYearsAgo, futureDate),
+  };
+
+  console.log('props.useSpice:', props.useSpice);
```
Comment (tyyzqmf) [inline src/reporting/quicksight-custom-resource.ts:111]: use logger.debug()

File: src/data-reporting-quicksight-stack.ts [hunk 4]
```diff
@@ -90,12 +90,12 @@ export class DataReportingQuickSightStack extends Stack {
...
 
-    const useTemplateArnCondition = new CfnCondition(
+    const useSpiceCondition = new CfnCondition(
       this,
-      'useTemplateArnCondition',
+      'useSpiceCondition',
       {
         expression:
-          Fn.conditionNot(Fn.conditionEquals(stackParams.quickSightTemplateArnParam.valueAsString, '')),
+          Fn.conditionEquals(stackParams.quickSightUseSpiceParam.valueAsString, 'yes'),
```
Comment (tyyzqmf) [inline src/data-reporting-quicksight-stack.ts:98]: Can we delete `quickSightTemplateArnParam` now?

File: src/reporting/quicksight-custom-resource.ts (hunk 3 above)
Comment (zxkane) [reply src/reporting/quicksight-custom-resource.ts:111]: It's fine to use `info` level logging for custom resource Lambda.

File: src/data-reporting-quicksight-stack.ts (hunk 4 above)
Comment (llmin) [reply src/data-reporting-quicksight-stack.ts:98]: It is a backup method for v1.0. In fact, we have already stopped the publishing of this ARN.

**Review by tyyzqmf — APPROVED** [review]

**Pull request merged**